llvm/lib/Transforms/Vectorize/VPlanRecipes.cpp

```diff
@@ -1965,8 +1965,11 @@ void VPWidenIntOrFpInductionRecipe::execute(VPTransformState &State) {
   }
 
   Value *SplatStart = Builder.CreateVectorSplat(State.VF, Start);
-  Value *SteppedStart = getStepVector(SplatStart, Step, ID.getInductionOpcode(),
-                                      State.VF, State.Builder);
+  assert(cast<VPInstruction>(getStepVector())->getOpcode() ==
```

Comment:
`&& step vector operand must be a StepVector` VPInstruction.

Although that may be too restrictive, for fixed vectors it could just be a constant vector?

Reply:
Fixed vectors would still be a VPInstruction::StepVector. Hopefully the need for this assertion would also go away if we fully expand the recipe in #118638